libraries/adaptive-expressions/src/builtinFunctions/string.ts

```diff
@@ -26,16 +28,31 @@ export class String extends ExpressionEvaluator {
      * @private
      */
     private static evaluator(): EvaluateExpressionDelegate {
-        return FunctionUtils.apply((args: any[]): string => {
-            if (typeof args[0] === 'string') {
-                return args[0];
+        return FunctionUtils.applyWithOptionsAndError((args: any[], options: Options): { value: any; error: string } => {
+            let result: any;
+            let error: string;
+            let locale = options.locale ? options.locale : Intl.DateTimeFormat().resolvedOptions().locale;
+            if (!error) {
+                locale = FunctionUtils.determineLocale(args, 2, locale);
             }
 
-            return JSON.stringify(args[0])
-                .replace(/(^\'*)/g, '')
-                .replace(/(\'*$)/g, '')
-                .replace(/(^\"*)/g, '')
-                .replace(/(\"*$)/g, '');
+            if (!error) {
+                if (typeof args[0] === 'string') {
+                    result = args[0];
+                } else if (typeof args[0] === 'number') {
+                    result = args[0].toLocaleString(locale);
+                } else if (args[0] instanceof Date) {
+                    result = args[0].toLocaleDateString(locale);
+                } else {
+                    result = JSON.stringify(args[0])
```

Comment:
A comment explaining the regular expressions might be helpful.

Reply:
added comments explaining the regular expressions